compute_sdk/globus_compute_sdk/sdk/web_client.py

```diff
@@ -55,15 +55,20 @@ def __init__(
         serializer: t.Optional[ComputeSerializer] = None,
     ):
         if function is not None:
-            function_name = function.__name__
-            function_code = _get_packed_code(function, serializer=serializer)
+            invalid = function_name is not None or function_code is not None
+        else:
+            invalid = function_name is None or function_code is None
 
-        if function_name is None or function_code is None:
+        if invalid:
             raise ValueError(
                 "Either 'function' must be provided, or "
-                "both of 'function_name' and 'function_code'"
+                "both of 'function_name' and 'function_code'."
             )
 
+        if function is not None:
+            function_name = function.__name__
+            function_code = _get_packed_code(function, serializer=serializer)
+
```

Comment:
I had a bit of a moment this morning wrapping my head around the "invalid" logic.  It's not incorrect, but the extra `invalid` indirection took me a minute to grok.  An alternative you might consider is to enforce directly the requirement that `function_name/code` are set:
```python
        if function is not None:
            if function_name or function_code:
                raise ValueError(...)
            function_name = function.__name__
            function_code =  _get_packed_code(function, serializer=serializer)

        if not function_code:
            raise ValueError("`function_code` not provided")
        if not function_name:
            raise ValueError("`function_name` not provided")
```

In my mind, the difference in approach is:
- `function_name` and `function_code` are required&nbsp;&mdash;&nbsp;the following code has that expectation&nbsp;&mdash;&nbsp;so make the check for them explicit and independent of each other
- beyond the bottom line (the first bullet point), the first if-condition "only" sets them if `function` is provided, and, as an along-the-way action, helps the user out (`raise`) if they provided ambiguous information.

Reply:
Fair point, I was trying to centralize the error call but different error messages would be helpful. Split that logic up